dao/src/test/java/se/fortnox/reactivewizard/db/DbProxyTest.java

```diff
@@ -45,12 +49,73 @@ public class DbProxyTest {
     MockDb mockDb;
     DbProxyTestDao dbProxyTestDao;
 
+    @Rule
+    public LoggingVerifier loggingVerifier = new LoggingVerifier(DbProxy.class);
+
+    @Test
+    public void shouldReturnDataFromDbForQueryWhenReturningMono() throws SQLException {
+        mockDb.addRowColumn(1, 1, "sql_val", String.class, "myname");
+        assertThat(dbProxyTestDao.select("mykey").block().getSqlVal()).isEqualTo("myname");
+        mockDb.verifySelect("select * from table where key=?", "mykey");
+
+        assertMockDbClosed();
+    }
+
     @Test
-    public void shouldReturnDataFromDbForQuery() throws SQLException {
+    public void shouldReturnDataFromDbForQueryWhenReturningFlux() throws SQLException {
         mockDb.addRowColumn(1, 1, "sql_val", String.class, "myname");
-        assertThat(dbProxyTestDao.select("mykey").toBlocking().single().getSqlVal()).isEqualTo("myname");
+        assertThat(dbProxyTestDao.selectFlux("mykey").blockLast().getSqlVal()).isEqualTo("myname");
         mockDb.verifySelect("select * from table where key=?", "mykey");
 
+        assertMockDbClosed();
+    }
+
+    @Test
+    public void shouldSignalErrorWhenFailingToCloseConnectionOnMonoReturnType() throws SQLException {
+        mockDb.addRowColumn(1, 1, "sql_val", String.class, "myname");
+        doThrow(new SQLException("close")).when(mockDb.getPreparedStatement()).close();
+        StepVerifier.create(dbProxyTestDao.select("myname")).verifyErrorMessage("close");
+    }
+
+    @Test
+    public void shouldSignalCompleteWhenQueryReturnsNoResultOnMonoReturnType() {
+        StepVerifier.create(dbProxyTestDao.selectMultipleWithMono()).verifyComplete();
+    }
+
+    @Test
+    public void shouldSignalErrorWhenDaoMethodDoesNotReturnAPublisher() {
+        try {
```

Comment:
use
`assertThatException()
            .describedAs("Expected RuntimeException with the message...")
            .isThrownBy(() -> )
            .withMessage(bla)`